app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -42,16 +44,23 @@ class PlantDetailFragment : Fragment() {
         val plantId = requireNotNull(arguments).getString(ARG_ITEM_ID)
 
         val factory = InjectorUtils.providePlantDetailViewModelFactory(requireActivity(), plantId)
-        val plantDetailViewModel = ViewModelProviders.of(this, factory)
-                .get(PlantDetailViewModel::class.java)
+        plantDetailViewModel = ViewModelProviders.of(this, factory)
+            .get(PlantDetailViewModel::class.java)
 
         val binding = DataBindingUtil.inflate<FragmentPlantDetailBinding>(
                 inflater, R.layout.fragment_plant_detail, container, false).apply {
             viewModel = plantDetailViewModel
             setLifecycleOwner(this@PlantDetailFragment)
             fab.setOnClickListener { view ->
```

Comment:
A better way here to implement `fab` click() is to use data-binding.
Like this:
`fragment_plant_detail.xml`
```
<fab
   onclick = "vm.toggleAddOrDelete()"
/>
```
`PlantDetailViewModel`:

```
fun toggleAddOrDelete() {
  if(..) delete()
  else add()
}
```

Reply:
Good, it link xml with ViewModel directly, I got it.